clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -514,6 +514,22 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                       << "marker type " << memref.marker.marker_type << " value 0x"
                       << std::hex << memref.marker.marker_value << std::dec << "\n";
         }
+        if (memref.marker.type == TRACE_TYPE_MARKER &&
+            memref.marker.marker_type == TRACE_MARKER_TYPE_KERNEL_EVENT &&
+            // TODO i#3937: We need to exclude this check for "kernel_xfer_app" when
+            // running online.
+            (knob_test_name_ != "kernel_xfer_app" || knob_offline_) &&
+            // All TRACE_MARKER_TYPE_RSEQ_ABORT markers are followd by
+            // TRACE_MARKER_TYPE_KERNEL_EVENT. We only want to consider the
+            // TRACE_MARKER_TYPE_KERNEL_EVENT markers that do not occur directly after
+            // an RSEQ abort marker.
+            !(shard->prev_entry_.data.type == TRACE_TYPE_MARKER &&
+              shard->prev_entry_.marker.marker_type == TRACE_MARKER_TYPE_RSEQ_ABORT)) {
+            const std::string pc_discontinuity_error_string =
+                check_for_pc_discontinuity(shard, memref, nullptr, false);
```

Comment:
The new checks for signals are also being done today as the "signal handler return point incorrect" check. Do we plan to remove the redundant checks?